CHANGES.rst

```diff
@@ -1,3 +1,9 @@
+2.6 (unreleased)
+----------------
+
+- Add a developer overview document to help understand how ASDF works
+  internally. Still a work in progress. [#729] 
```

Comment:
The PR number is `730`.

Reply:
OK, I put the the issue number (a bit of chicken and egg thing since one can't be sure of the PR number until it is submitted, can one?)